hypothesis-python/tests/ghostwriter/test_expected_output.py

```diff
@@ -0,0 +1,133 @@
+# This file is part of Hypothesis, which may be found at
+# https://github.com/HypothesisWorks/hypothesis/
+#
+# Most of this work is copyright (C) 2013-2020 David R. MacIver
+# (david@drmaciver.com), but it contains contributions by others. See
+# CONTRIBUTING.rst for a full list of people who may hold copyright, and
+# consult the git log if you need to determine who owns an individual
+# contribution.
+#
+# This Source Code Form is subject to the terms of the Mozilla Public License,
+# v. 2.0. If a copy of the MPL was not distributed with this file, You can
+# obtain one at https://mozilla.org/MPL/2.0/.
+#
+# END HEADER
+
+"""
+'Golden master' tests for the ghostwriter.
+
+To update the recorded outputs, simply run the tests with the
+"GHOSTWRITER_UPDATE" environment variable set.
+"""
+
+import os
+import re
+from typing import Sequence
+
+import pytest
+
+from hypothesis.extra import ghostwriter
+
+
+def timsort(seq: Sequence[int]) -> Sequence[int]:
+    return sorted(seq)
+
+
+# Note: for some of the `expected` outputs, we replace away some small
+#       parts which vary between minor versions of Python.
+@pytest.mark.parametrize(
+    "data",
+    [
+        ("FUZZ_SORTED", ghostwriter.fuzz(sorted)),
+        ("RE_COMPILE", ghostwriter.fuzz(re.compile)),
+        (
+            "RE_COMPILE_EXCEPT",
+            ghostwriter.fuzz(re.compile, except_=re.error)
+            # re.error fixed it's __module__ in Python 3.7
+            .replace("import sre_constants\n", "").replace("sre_constants.", "re."),
+        ),
+        ("RE_COMPILE_UNITTEST", ghostwriter.fuzz(re.compile, style="unittest")),
+    ],
+    ids=lambda x: x[0],
+)
+def test_ghostwriter_example_outputs(data):
+    name, actual = data
+    expected = globals()[name][1:]  # remove leading newline
+    if "GHOSTWRITER_UPDATE" in os.environ:
+        print("Updating", name)
+        with open(__file__) as f:
```

Comment:
I feel like having these as separate files that are stored in the repository would be a lot cleaner and that attempting to rewrite Python files is going to prove to be a maintenance headache.

Reply:
The problem is that we then instead have to configure all our linters and formatters to avoid looking at the example outputs, and remember that the build system doesn't work at all on Windows.

It's dumb code but it works just fine!

Reply:
> The problem is that we then instead have to configure all our linters and formatters to avoid looking at the example outputs

I was assuming you'd give them a `.txt` extension or something, which is all the configuration that needs.

> remember that the build system doesn't work at all on Windows.

The build system works fine on Windows, you just have to use WSL2 (which is what I do), but also I'm not sure why this matters?

Reply:
:facepalm:, of course they don't need a `.py` extension.  I'll push that shortly.

For most of the sprints and mentoring stuff I've done, new contributors have had a pretty hard time getting set up and often just run pytest and linters by hand.  Complicated config would make this a lot harder, though as you say we don't actually need it!